src/client/renderer/r_state.c

```diff
@@ -434,13 +459,52 @@ void R_CreateBuffer(r_buffer_t *buffer, const GLenum hint, const r_buffer_type_t
 	if (size) {
 		R_UploadToBuffer(buffer, size, data);
 	}
+
+	r_state.buffers_total++;
+}
+
+/**
+ * @brief Allocate an interleaved GPU buffer of the specified size.
+ * Optionally upload the data immediately too.
+ */
+void R_CreateInterleaveBuffer(r_buffer_t *buffer, const GLubyte struct_size, const r_buffer_layout_t *layout,
+                              const GLenum hint, const size_t size, const void *data) {
+
+	R_CreateBuffer(buffer, R_ATTRIB_TOTAL_TYPES, struct_size, false, hint, R_BUFFER_DATA | R_BUFFER_INTERLEAVE, size, data);
+
+	GLsizei stride = 0;
+
+	for (; layout->attribute != -1; layout++) {
+		stride += layout->size;
+		buffer->interleave_attribs[layout->attribute] = layout;
+		buffer->attrib_mask |= 1 << layout->attribute;
+
+		// init type pack state once
+		if (!layout->_type_state.packed) {
+			r_buffer_layout_t *temp = (r_buffer_layout_t *) layout;
+
+			temp->_type_state.type = layout->type;
+			temp->_type_state.stride = buffer->element_type.stride;
+			temp->_type_state.offset = layout->offset;
+			temp->_type_state.count = layout->count;
+			temp->_type_state.normalized = layout->normalized;
+		}
+	}
+
+	if (stride != struct_size) {
+		Com_Error(ERR_DROP, "Buffer interleave size doesn't match layout size\n");
+	}
 }
 
 /**
  * @brief Destroy a GPU-allocated buffer.
  */
 void R_DestroyBuffer(r_buffer_t *buffer) {
-	assert(buffer->bufnum != 0);
+
+	if (buffer->bufnum == 0) {
+		Com_Warn("Attempting to free empty buffer");
```

Comment:
A Com_Debug would be fine here. This is a development bug trap, and I typically only use Warn for things a user might see under normal operation.